File: src/com/Torvald/Terrarum/UserInterface/UIClickable.java
package com.Torvald.Terrarum.UserInterface;

/**
 * Created by minjaesong on 16-03-06.
 */
public interface UIClickable {

    void mouseMoved(int oldx, int oldy, int newx, int newy);

    void mouseDragged(int oldx, int oldy, int newx, int newy);

    void mousePressed(int button, int x, int y);

    void mouseReleased(int button, int x, int y);

    void mouseWheelMoved(int change);

    void controllerButtonPressed(int controller, int button);

    void controllerButtonReleased(int controller, int button);

}

File: src/com/Torvald/Terrarum/UserInterface/UIClickable.kt
package com.Torvald.Terrarum.UserInterface

/**
 * Created by minjaesong on 16-03-14.
 */
interface UIClickable {
    fun mouseMoved(oldx: Int, oldy: Int, newx: Int, newy: Int)

    fun mouseDragged(oldx: Int, oldy: Int, newx: Int, newy: Int)

    fun mousePressed(button: Int, x: Int, y: Int)

    fun mouseReleased(button: Int, x: Int, y: Int)

    fun mouseWheelMoved(change: Int)

    fun controllerButtonPressed(controller: Int, button: Int)

    fun controllerButtonReleased(controller: Int, button: Int)

}